core/src/com/unciv/logic/map/UnitMovementAlgorithms.kt

```diff
@@ -276,20 +275,26 @@ class UnitMovementAlgorithms(val unit:MapUnit) {
      * DOES NOT designate whether we can reach that tile in the current turn
      */
     fun canMoveTo(tile: TileInfo): Boolean {
-        if(unit.type.isAirUnit())
-            if(tile.isCityCenter())
-                return tile.airUnits.size<6 && tile.getCity()?.civInfo==unit.civInfo
-            else if(tile.militaryUnit!=null) {
-                val unitAtDestination = tile.militaryUnit!!
-
-                var unitCapacity = if (unitAtDestination.getUniques().contains("Can carry 2 aircraft")) 2 else 0
-                // unitCapacity += unitAtDestination.getUniques().count { it == "Can carry 1 extra air unit" }
-
-                return ((unitAtDestination.type.isAircraftCarrierUnit() && !unit.type.isMissileUnit()) ||
-                        (unitAtDestination.type.isMissileCarrierUnit() && unit.type.isMissileUnit()))
-                        && unitAtDestination.owner==unit.owner && tile.airUnits.size < unitCapacity
-            } else
-                return false
+        if(unit.type.isAirUnit()) {
+            if (tile.isCityCenter()) {
+                    if (tile.airUnits.filter { !it.isTransported }.size < 6 && tile.getCity()?.civInfo == unit.civInfo)
+                        return true // if city is free - no problem, get in
+                    if (!unit.isTransported)
+                        return false // if no space and it is not on carrier - get out
+                } // let's check whether it enters city on carrier now...
+
+                if (tile.militaryUnit != null) {
+                    val unitAtDestination = tile.militaryUnit!!
+
+                    var unitCapacity = if (unitAtDestination.getUniques().contains("Can carry 2 aircraft")) 2 else 0
+                    // unitCapacity += unitAtDestination.getUniques().count { it == "Can carry 1 extra air unit" }
+
+                    return ((unitAtDestination.type.isAircraftCarrierUnit() && !unit.type.isMissileUnit()) ||
+                            (unitAtDestination.type.isMissileCarrierUnit() && unit.type.isMissileUnit()))
+                            && unitAtDestination.owner == unit.owner && tile.airUnits.filter { it.isTransported }.size < unitCapacity
+                } else
+                    return false
```

Comment:
Please put all of this into a separate function, airUnitCanMoveTo() or something

Reply:
Thank you for the feedback. Done.